src/webots/nodes/WbInertialUnit.cpp

```diff
@@ -126,8 +126,10 @@ void WbInertialUnit::computeValue() {
   // get north and -gravity in global coordinate systems
   const WbWorldInfo *const wi = WbWorld::instance()->worldInfo();
   const WbVector3 &north = wi->northDirection().normalized();
-  const WbVector3 &minusGravity = -wi->gravity().normalized();
+  WbVector3 minusGravity = -wi->gravity().normalized();
 
+  if (minusGravity.isNan())
+    minusGravity = WbVector3(0.0, 1.0, 0.0);
```

Comment:
Shouldn't we set it to (0, 0, 0) instead?

Reply:
No otherwise the rotation matrix computed just after will not be valid and roll pitch yaw will probably always be 0.

Reply:
OK, but did you check that the resulting IMU values computed from this arbitrary unit vector are correct? If we set it to (1.0, 0.0, 0.0) for example, do we get the same result?

Reply:
We will not get the same result with `1.0, 0.0, 0.0` but the `0.0, 1.0, 0.0` vector was chosen to have X=roll, Z=pitch, Y=yaw, which is the default behavior and corresponds to what is displayed in the doc (see this image for example: https://cyberbotics.com/doc/reference/inertialunit#field-summary ).

Reply:
Yes, but if `WorldInfo.northDirection` is `0 1 0`, then it will not work...
I believe that `northDirection` and `gravity` should be orthogonal and this constraint is not checked by Webots.
To better model this, we should replace the `WorldInfo.northDirection` and `gravity` vectors with a `northOrientation` rotation and a `gravity` scalar value.

Reply:
Let's go ahead with this PR as it is (although it looks wrong to me). I will open an issue about a better way to indicate the north direction and gravity in the `WorldInfo` node.

Reply:
Ok so I am merging right now then.

Reply:
I agree, there is definitely something going wrong with the `gravity` and `northDirection` (at least in the context of the inertialUnit), but this PR fixes this the original crash and works now fine except in the case where `gravity` and `northDirection` are parallel.